sqlx-core/src/sqlite/connection/mod.rs

```diff
@@ -146,6 +151,14 @@ impl Connection for SqliteConnection {
 
     fn close(mut self) -> BoxFuture<'static, Result<(), Error>> {
         Box::pin(async move {
+            if self.optimize_on_close {
+                let mut pragma_string = String::new();
+                if let Some(limit) = self.analysis_limit {
+                    pragma_string.push_str(&format!("PRAGMA analysis_limit = {}; ", limit));
```

Comment:
I believe @paolobarbolini only meant that you should use `.push_str()` instead of `write!()` on line 159; `write!()` is more appropriate here because it's adding formatted content directly to the string.